### PR TITLE
feat: auto wallet provision

### PR DESCRIPTION
**Pull request by samsiegart**
refs https://github.com/Agoric/dapp-inter/issues/214

## Screenshot
 
Screenshot 2024-01-02 at 12 59 57 AM

### CODE REVIEW

File: src/components/ProvisionSmartWalletNoticeDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  const smartWalletFeeForDisplay = smartWalletFee
+    ? smartWalletFee + ' IST'
+    : null;
+
+  const body = (
+    <span>
+      To interact with contracts on the Agoric chain, a smart wallet must be
+      created for your account. As an anti-spam measure, you will need{' '}
+      {smartWalletFeeForDisplay && <b>{smartWalletFeeForDisplay}</b>} to fund
+      its provision which will be deposited into the community fund. Click
```
Comment (mhofman) [inline src/components/ProvisionSmartWalletNoticeDialog.tsx:61]: I'm rusty on this, is the "community fund" the same as the "reserve pool"?

I don't remember where explicit provision fees ended up in (I think also in the reserve pool). The new auto-provision mechanism charges like other swingset fees, and it ends up in the reserve pool.

**Review by mhofman — COMMENTED** [review]
FYI, while 1 IST is charged for provisioning, 0.25 IST is still transferred from the provision pool to the user's account during provisioning, which means that effectively, only 0.75 IST will be deducted (but 1 IST is still needed). I don't think we want to go into that much details however.

**Comment by otoole-brendan** [comment]
@samsiegart per recent Slack conversation - let's remove reference to community fund and change to

'....which will be deposited into the reserve pool.'

**Comment by samsiegart** [comment]
> @samsiegart per recent Slack conversation - let's remove reference to community fund and change to
> 
> '....which will be deposited into the reserve pool.'

Fixed

**Review by 0xpatrickdev — APPROVED** [review]

**Pull request merged**